drivers/modem/Kconfig.gsm

```diff
@@ -34,6 +34,13 @@ config MODEM_GSM_QUECTEL
 
 endchoice
 
+config MODEM_GSM_PTP_CLOCK
```

Comment:
This looks better now but actually the GSM modem most likely cannot really provide PTP (Precision Time Protocol) support which is typically only implemented for Ethernet. I was suggesting earlier that something similar like what ptp_clock driver is implemented, not exactly that we directly use ptp_clock. For example for modem clock driver, there is no need to have support for adjust() or rate_adjust(), also probably the set() is also not needed (we most probably cannot set the time in the modem, just read what the current time is).

Reply:
> the GSM modem most likely cannot really provide PTP

Yeah, I'm literally hijacking the ptp driver API here. Not really sure what is the best path forward.

>Currently there is no public driver API for modems so we would need to create one, and then perhaps integrate this RTC (time/date) and GNSS (location) functionality to that API. Then the wall clock time API functions could use the modem driver API to fetch the current time in a generic way.

I recently worked with the esp8266 wifi driver and needed the same function, maybe we need an AT driver API, but I'm not sure